runtime/astar/src/lib.rs

```diff
@@ -1118,20 +1169,44 @@ pub type Executive = frame_executive::Executive<
 >;
 
 parameter_types! {
-    pub const DappStakingMigrationName: &'static str = "DappStakingMigration";
-
+    pub const StaticPriceProviderName: &'static str = "StaticPriceProvider";
 }
 
 /// All migrations that will run on the next runtime upgrade.
 ///
 /// Once done, migrations should be removed from the tuple.
 pub type Migrations = (
     frame_support::migrations::RemovePallet<
-        DappStakingMigrationName,
+        StaticPriceProviderName,
         <Runtime as frame_system::Config>::DbWeight,
     >,
+    OracleIntegrationLogic,
+    pallet_price_aggregator::PriceAggregatorInitializer<Runtime, InitPrice>,
 );
 
+pub struct InitPrice;
+impl Get<CurrencyAmount> for InitPrice {
+    fn get() -> CurrencyAmount {
+        // 0.18 $
+        CurrencyAmount::from_rational(18, 100)
+    }
+}
+
+use frame_support::traits::OnRuntimeUpgrade;
+pub struct OracleIntegrationLogic;
+impl OnRuntimeUpgrade for OracleIntegrationLogic {
+    fn on_runtime_upgrade() -> Weight {
+        // 1. Set initial storage versions for the membership pallet
+        use frame_support::traits::StorageVersion;
+        StorageVersion::new(4)
+            .put::<pallet_membership::Pallet<Runtime, OracleMembershipInstance>>();
+
```

Comment:
dumb question: Why do we need to override storage version for membership here?

Reply:
We're not overriding it, we're setting it. Otherwise it would be 0 AFAIK.

Reply:
Ok that make sense